### ISSUE TITLE
Very small LoRA adapter size training Qwen3.5 MoE models

### ISSUE THREAD

**Issue by Nero10578** [axolotl-ai-cloud/axolotl#3514]
### Please check that this issue hasn't been reported before.

- [x] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

LoRA adapter should contain all the important targets and be an expected size for the rank. 

### Current behaviour

The resulting LoRA adapters for Qwen3.5 models results in extremely small adapter files when using the Qwen3.5 MoE example train configs.

For example Qwen 3.5 35B LoRA rank 16 only results in a 13.5mb adapter.safetensors file.

### Steps to reproduce

Train any Qwen3.5 MoE models with the example config and you will see the resulting adapter size is abnormally small and cannot possibly be correct.

### Config yaml

```yaml
base_model: /home/arli/models/Qwen3.5-35B-A3B-Derestricted

plugins:
  - axolotl.integrations.cut_cross_entropy.CutCrossEntropyPlugin
cut_cross_entropy: true

load_in_8bit: false
load_in_4bit: false
quantize_moe_experts: false

datasets:
  - path: /home/arli/train/datasets/nothink.jsonl
    type: chat_template

    field_messages: conversations
    message_property_mappings:
      role: from
      content: value

dataset_prepared_path: ./last_run_prepared
val_set_size: 0
output_dir: ./outputs/lora-out-minLR-v1

adapter: lora
save_safetensors: true

sequence_len: 16384
sample_packing: true
pad_to_sequence_len: true

lora_r: 16
lora_alpha: 32
lora_dropout: 0.0
lora_target_modules:
  - q_proj
  - k_proj
  - v_proj
  - o_proj
  
lora_target_parameters:
  - mlp.experts.gate_up_proj
  - mlp.experts.down_proj

lora_mlp_kernel: false
lora_qkv_kernel: false
lora_o_kernel: false

gradient_accumulation_steps: 8
micro_batch_size: 2
num_epochs: 1
optimizer: adamw_torch_fused
lr_scheduler: constant
learning_rate: 0.000001
max_grad_norm: 0.5

bf16: auto

wandb_project: qwen-3.5-35b-d

auto_resume_from_checkpoints: false
logging_steps: 1
flash_attention: true

warmup_ratio: 0
evals_per_epoch: 0
saves_per_epoch: 4
save_total_limit: 4

gradient_checkpointing: false
gradient_checkpointing_kwargs:
  use_reentrant: false

fsdp_config:
  fsdp_version: 2
  offload_params: true
  cpu_ram_efficient_loading: false
  auto_wrap_policy: TRANSFORMER_BASED_WRAP
  transformer_layer_cls_to_wrap: Qwen3_5MoeDecoderLayer
  state_dict_type: FULL_STATE_DICT
  sharding_strategy: FULL_SHARD
  reshard_after_forward: true
  activation_checkpointing: true

context_parallel_size: 1
```

### Possible solution

It seems setting lora_target_modules using a regex manages to fix the adapters and results in adapter file sizes that's expected for a model of the size compared to other MoE models. It also works with vllm using this PR https://github.com/vllm-project/vllm/pull/36976.

`lora_target_modules: 'model\.(language_model\.)?layers\.[\d]+\.(mlp|self_attn)\.(shared_expert\.)?(up|down|gate|gate_up|q|k|v|o)_proj'`

However, I am not sure if this should be the final fix or if there would be a better solution.

### Which Operating Systems are you using?

- [x] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.12

### axolotl branch-commit

main/163bd4dd5a9dc6097d923cdd733cf5d1593c056c

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this bug has not been reported yet.
- [x] I am using the latest version of axolotl.
- [x] I have provided enough information for the maintainers to reproduce and diagnose the issue.
